trieste/acquisition/function.py

```diff
@@ -557,16 +548,18 @@ def __init__(
         :param constraint_builder: The builder for the constraint function.
         :param min_feasibility_probability: The minimum probability of feasibility for a
             "best point" to be considered feasible.
-        :raise ValueError (or InvalidArgumentError): If ``min_feasibility_probability`` is not a
-            scalar in the unit interval :math:`[0, 1]`.
+        :raise ValueError (or tf.errors.InvalidArgumentError): If ``min_feasibility_probability``
+            is not a scalar in the unit interval :math:`[0, 1]`.
         """
         tf.debugging.assert_scalar(min_feasibility_probability)
 
-        if not 0 <= min_feasibility_probability <= 1:
-            raise ValueError(
-                f"Minimum feasibility probability must be between 0 and 1 inclusive,"
-                f" got {min_feasibility_probability}"
-            )
+        if isinstance(min_feasibility_probability, (int, float)):
+            tf.debugging.assert_greater_equal(float(min_feasibility_probability), 0.0)
+            tf.debugging.assert_less_equal(float(min_feasibility_probability), 1.0)
+        else:
+            dtype = min_feasibility_probability.dtype
+            tf.debugging.assert_greater_equal(min_feasibility_probability, tf.cast(0.0, dtype))
+            tf.debugging.assert_less_equal(min_feasibility_probability, tf.cast(1.0, dtype))
```

Comment:
i'm aware this is very verbose. The difficulty is that `0` is a `float` statically but not at runtime so a user can pass 0 and it pass mypy but be interpreted as an integer at runtime